src/test/java/com/example/java/test/junior/developer/QuestionMapperTest.java

```diff
@@ -0,0 +1,46 @@
+package com.example.java.test.junior.developer;
+
+import com.example.java.test.junior.developer.dto.QuestionDto;
+import com.example.java.test.junior.developer.mapper.QuestionMapper;
+import com.example.java.test.junior.developer.model.Question;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+public class QuestionMapperTest {
+
+    @Mock
+    private QuestionMapper questionMapper;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        questionMapper = new QuestionMapper();
+    }
+
+    @Test
+    public void toEntity_ValidDto_ReturnsEntity() {
+        QuestionDto questionDto = new QuestionDto(0, "What is polymorphism?");
+
+        Question result = questionMapper.toEntity(questionDto);
+
+        assertEquals(questionDto.getName(), result.getName());
+        assertEquals(questionDto.getId(), result.getId());
```

Comment:
This test showed us mistake that you made. Entity id cannot be primitive. Is it correct that ALL entities will have 0 id? What will Hibernate do? It will try to update it. And fail because there is no record in DB with id 0. 
You should make id - Integer or Long.